snuba/web/query.py

```diff
@@ -50,6 +55,32 @@
 MAX_QUERY_SIZE_BYTES = 256 * 1024  # 256 KiB by default
 
 
+def _run_new_query_pipeline(
+    request: Request,
+    timer: Timer,
+    query_metadata: SnubaQueryMetadata,
+    robust: bool = False,
+    concurrent_queries_gauge: Optional[Gauge] = None,
+) -> QueryResult:
+    clickhouse_query = EntityAndStoragePipelineStage().execute(
+        QueryPipelineResult(
+            data=request, query_settings=request.query_settings, timer=timer, error=None
+        )
+    )
+    res = ExecutionStage(
+        request.attribution_info,
+        query_metadata=query_metadata,
+        robust=robust,
+        concurrent_queries_gauge=concurrent_queries_gauge,
+    ).execute(clickhouse_query)
+    if res.error:
+        raise res.error
+    elif res.data:
+        return res.data
+    # we should never get here
+    raise Exception("No result or data, very bad exception")
```

Comment:
Voila, the new pipeline. 

Pipeline chaining can probably be made to look nicer, but we can save that for another time